src/core/api/link_raw_api.cpp

```diff
@@ -228,9 +228,13 @@ otError otLinkRawSetMacKey(otInstance *    aInstance,
                            const otMacKey *aCurrKey,
                            const otMacKey *aNextKey)
 {
+    const otMacKeyMaterial *aPrevKeyMaterial = reinterpret_cast<const otMacKeyMaterial *>(aPrevKey);
+    const otMacKeyMaterial *aCurrKeyMaterial = reinterpret_cast<const otMacKeyMaterial *>(aCurrKey);
+    const otMacKeyMaterial *aNextKeyMaterial = reinterpret_cast<const otMacKeyMaterial *>(aNextKey);
+
     return static_cast<Instance *>(aInstance)->Get<Mac::LinkRaw>().SetMacKey(
-        aKeyIdMode, aKeyId, *static_cast<const Mac::Key *>(aPrevKey), *static_cast<const Mac::Key *>(aCurrKey),
-        *static_cast<const Mac::Key *>(aNextKey));
+        aKeyIdMode, aKeyId, *static_cast<const Mac::Key *>(aPrevKeyMaterial),
+        *static_cast<const Mac::Key *>(aCurrKeyMaterial), *static_cast<const Mac::Key *>(aNextKeyMaterial));
```

Comment:
@abtink As I understand, `link_raw` module is used for RCP and NCP applications. And for these applications we cannot use KeyRefs as the idea of key refs is that they keys are stored in secure storage and only the references are used in the stack. As the keys are stored in host, RCP can't use these keys even with references available. That is why I have not extended the KeyRefs for RCPs and hence to link_raw.

Reply:
Yes. that's correct that RCP uses `LinkRaw` but `LinkRaw` feature itself can be enabled and used in other models (e.g we used to have an internal project where raw link was used on SoC style device, a sleepy end-device everything running on the same radio chip). 

I think (at some point) we may need to support PSA on an RCP. 
- In this case, I expect that the keys are passed down to RCP as literal, but then RCP itself would save them in PSA ITS and not in RAM.

Here are the options I can suggest here:
1) We can leave adding support for `OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE` when link-raw is also enabled for a future PR.
   - In this case, I think we need to add check here and give an build error if this combination of config is being used.
   - We don't want it to build but not work correctly.
2) We can add support for it and handle this combination
   - We can assume the `LinkRaw` API gets the key as literal key `otMacKey`
   - We can then save them in ITS and convert them to `KeyMaterial` (if we use the helper method `Mac::KeyMaterial::SetFrom(const Mac::Key &)` from https://github.com/openthread/openthread/pull/6862#discussion_r692434241 I think it will be relatively easy to do)

Reply:
@abtink Yes, I am on board for the second Idea, and that why I did not add any errors. My idea was when building RCP, the platform abstraction for radio needs to handle receiving the key as a literal and saving it in PSA, also using the same for Enh Ack handling.

So my question is, Is there a Posix based RCP that we need to fix? Because if not, I think the stack is working as expected right? Because as I see it, Link Raw is connected to spinel to send the key to radio, which we still want to do, and the import needs to happen on RCP after it receives the literal key, which I think can only be done in radio.c.

Reply:
Never mind, found a way 😃

Reply:
We have two components in RCP mode. 
- The RCP firmware itself, which uses the `OPENTHREAD_RADIO` config.  I expect the change to support PSA for this  to be in `LinkRaw` (in the `LinkRaw::SendMacKey()` where we get the key and need to save is ITS and then inform `SubMac`, etc).
- The RCP host driver component which is posix code (`radio_spinel`, and related `posix` module in OT). I think the change there is that when we talk to RCP (setting key on RCP or equivalent to  invoking the `SetMacKey()` API call in `RadioSpinel`), we should ensure to send the literal key down to RCP (if we have key ref we need to export the literal key).

Reply:
Yup, found that and added the required change.